pkg/resolution/resolver/hub/resolver.go

```diff
@@ -107,14 +136,19 @@ func (r *Resolver) Resolve(ctx context.Context, params []pipelinev1beta1.Param)
 		paramsMap[p.Name] = p.Value.StringVal
 	}
 
-	if _, ok := paramsMap[ParamCatalog]; !ok {
-		if catalogString, ok := conf[ConfigCatalog]; ok {
-			paramsMap[ParamCatalog] = catalogString
+	// type
+	if _, ok := paramsMap[ParamType]; !ok {
+		if typeString, ok := conf[ConfigType]; ok {
+			paramsMap[ParamType] = typeString
 		} else {
-			return nil, fmt.Errorf("default catalog was not set during installation of the hub resolver")
+			return nil, fmt.Errorf("default type was not set during installation of the hub resolver")
 		}
 	}
+	if paramsMap[ParamType] != ArtifactHubType && paramsMap[ParamType] != TektonHubType {
+		return nil, fmt.Errorf("type param must be artifact or tekton")
+	}
```

Comment:
I feel like we might be able to extract a helper function from `ValidateParams` and `Resolve` regarding params i.e. [one good example from gitresolver](https://github.com/tektoncd/pipeline/blob/7189162fd3ff5d9aa11b7090134aab5566f1cfbc/pkg/resolution/resolver/git/resolver.go#L406) and this helper function does 3 things: 1/. populate the params with default from configmap and from user-provided spec, 2/. validate param values, 3/. find missing params. 

This can bring 2 benefits to us 1/. make sure we do the 3 things mentioned above consistently throughout both `ValidateParams` and `Resolve` functions. 2/. collect all the missing params together and tell users only once the names of all missing params instead of finding one missed -> return error.

Reply:
Good point. I agree the point that we can returning all missing values in one error message, that definitely helps users. I get the point you want to do the mutation first before validation, but I'm not sure if doing the three things in 1 helper function really helps (as I personally don't expect value population functionality in a `validate()` helper method).

I can refactor this part to 2 helper functions: `validate()` and `populateDefaultParams()`? And we can also apply the same idea in git resolver later separately.

Reply:
SGTM! 2 helpers will make things even more clear. 👍 

echo what you said - I couldn't tell that the populateDefaultParams in gitresolver does two jobs until I saw the implementation returns validation error.

Reply:
> echo what you said - I couldn't tell that the populateDefaultParams in gitresolver does two jobs until I saw the implementation also returns validation error.

Yeah, that's my bad - I'm prone to seeing code that does much of the work for two different things and just combining the code a bit more often than I should be. =)

Reply:
@abayer NOT A PROBLEM AT ALL! It's still a very good example for how to reuse code for other resolvers 👍 !!! 😄

Reply:
@abayer, @chuangw6. I refactored the code based on discussion here, PTAL 😁